source/usage-examples/find.txt

```diff
@@ -0,0 +1,69 @@
+=======================
+Find Multiple Documents
+=======================
+
+.. default-domain:: mongodb
+
+You can query for multiple documents in a collection by calling the ``find()``
+method on a ``MongoCollection`` object. Pass a query(TODO: link to filter builder) filter to the
+``find()`` method to query for and return documents that match the filter in
+the collection. If you do not include a filter, MongoDB returns all the
+documents in the collection.
+
+For more information on querying MongoDB with the Java driver, see our
+guide (TODO: link) to querying documents.
+
+You can also chain methods to the ``find()`` method such as ``sort()``
+(TODO: link) which organizes the matched documents in a specified order and
+``projection()`` (TODO: link) which configures the included fields in the
+return document.
+
+The ``find()`` method returns an instance of ``FindIterable``, a class
+that offers several methods to access, organize, and traverse the results.
+``FindIterable`` also inherits methods from its parent class,
+``MongoIterable`` which implements the core Java interface ``Iterable``.
+
+If you call the ``iterator()`` method on the ``MongoIterable``, it
+returns a ``MongoCursor`` which you can use to traverse the results.
+You can call methods on the ``MongoCursor`` such as ``hasNext()`` to check
+whether additional results exist, or ``next()`` to return the next document
+in the collection. If no documents match the query, calling ``hasNext()``
+returns ``false`` and therefore calling ``next()`` throws an exception.
+
+For additional information on the aforementioned classes and methods, see the
+following API documentation:
+
+- :java-sync-api:`FindIterable <com/mongodb/client/FindIterable.html>`
+- :java-sync-api:`MongoIterable <com/mongodb/client/MongoIterable.html>`
+- :java-sync-api:`MongoCursor <com/mongodb/client/MongoCursor.html>`
+- :java-sync-api:`find() <com/mongodb/client/MongoCollection.html#find()>`
+
+If you call ``next()`` on the iterator either after it has returned the final
+result or when no results exist, it throws an exception of type
+``java.util.NoSuchElementException``. Always use ``hasNext()`` to check that
+additional results exist before every time you call ``next()``.
+
+
+Example
+-------
+
+The following snippet finds and prints all documents that match a query on
+the ``movies`` collection. It uses the following objects and methods:
+
+- A **query filter** that is passed in the ``find()`` method. The `lt()``
+  filter matches only movies with a runtime of less than 15 minutes.
+
+- A **sort** that organizes returned documents in descending order by
+  title ("Z" before "A").
+
+- A **projection** that includes the objects in the ``title`` and ``imdb``
+  fields and excludes the ``_id`` field using the helper method
+  ``excludeId()``.
+
+.. include:: /includes/connect-guide-note.rst
+
+.. literalinclude:: /includes/usage-examples/code-snippets/Find.java
```

Comment:
Are there plans to demonstrate the `MongoIterable.forEach` or `MongoIterable.into` helpers?  They are more user friendly than the iterator API but come at the cost of consuming all the elements from the cursor.

Reply:
I can make sure to include both in the Fundamentals > CRUD Guide > Read Operations unless you think they're essential for the Jasmine persona to see multiple ways when looking for an executable example -- I'm indifferent here as it would make the example longer, but would contain useful information for anyone that decided not to do further reading in the Fundamentals section. Let me know what you think!

For reference, here's the planned structure (see column E to H for section info):
https://docs.google.com/spreadsheets/d/1gFGQHrk-I9HyFbkqt7EmHmFymlmF4dSND1rgBhGP0eI/edit#gid=0

Reply:
Sounds good